reframe/core/pipeline.py

```diff
@@ -622,6 +622,11 @@ class RegressionTest(metaclass=RegressionTestMeta):
     #:    A new more powerful syntax was introduced
     #:    that allows also custom job script directive prefixes.
     #:
+    #: .. note::
+    #:    .. versionchanged:: 2.22
+    #:
+    #:    The attr:`extra_resources` is now used during the generation of
+    #:    allowing its definition after the :func:`setup` phase.
```

Comment:
I don't think that this clarification is needed, since nothing relevant was mentioned in the documentation. This is an implementation detail.